### PR TITLE
Handle region/regions when looking upstream for amiDetails

### PR DESCRIPTION
**Pull request by ajordens**
- Also `amiDetails` is a List<Map> (not sure how this ever worked previously)

### CODE REVIEW

**Comment by ajordens** [comment]
PTAL @spinnaker/netflix-reviewers

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy [hunk 1]
```diff
@@ -35,8 +35,11 @@ trait DeploymentDetailsAware {
     def result = [:]
     if (previousStage) {
       if (previousStage.context.containsKey("amiDetails")) {
-        result.amiName = previousStage.context.amiDetails.ami
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy]: Not sure how this ever worked previously ... amiDetails is a `List<Map>` and `amiDetails.ami` would have resolved to a `List<String>`.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy [hunk 2]
```diff
@@ -47,7 +50,8 @@ trait DeploymentDetailsAware {
 
   Stage getPreviousStageWithImage(Stage stage, String targetRegion) {
     getAncestors(stage).find {
-      (it.context.containsKey("ami") || it.context.containsKey("amiDetails")) && it.context.region == targetRegion
+      def regions = (it.context.region ? [it.context.region] : it.context.regions) as Set<String>
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy]: Similarly here ... would have never matched a `FindImage` stage which has `regions` vs. `region`

**Comment by anotherchrisberry** [comment]
Looks reasonable to me.

**Pull request merged**